<a href="https://colab.research.google.com/github/nurfnick/Cantrell_Jackson/blob/master/Truck_Type_Organization.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pa
import numpy as np

In [2]:
from google.colab import files
uploaded = files.upload()

Saving trucks.csv to trucks.csv


In [0]:
import io
df2 = pa.read_csv(io.BytesIO(uploaded['trucks.csv']))# Dataset is now stored in a Pandas Dataframe

In [0]:
df2

id  pri_id  hs_customer  yard  ...  tag color licenseplatenumber  status
0        1     739            1   1.0  ...  NaN   NaN                NaN  active
1        2     740            1   1.0  ...  NaN   NaN                NaN  active
2        3     741            1   1.0  ...  NaN   NaN                NaN  active
3        4       0            2   3.0  ...  NaN   NaN                NaN  active
4        5       0            2   3.0  ...  NaN   NaN                NaN  active
...    ...     ...          ...   ...  ...  ...   ...                ...     ...
3870  3871       0           19  43.0  ...  NaN   NaN                NaN  active
3871  3872       0           19  43.0  ...  NaN   NaN                NaN  active
3872  3873       0           19  43.0  ...  NaN   NaN                NaN  active
3873  3874       0           19  43.0  ...  NaN   NaN                NaN  active
3874  3875       0           19  43.0  ...  NaN   NaN                NaN  active

[3875 rows x 14 columns]

In [0]:
count=df2['truck_type'].value_counts()
print(count)
#number of truck_types

0      431
121    306
154    283
127    254
103    181
      ... 
181      1
58       1
118      1
9        1
174      1
Name: truck_type, Length: 169, dtype: int64


In [0]:
df2[df2['truck_type']==0]

#Truck Type 0 active and inactive


id  pri_id  hs_customer   yard  ...  tag color licenseplatenumber  status
2158  2159       0           18   39.0  ...  NaN   NaN                NaN  active
2160  2161       0           18   37.0  ...  NaN   NaN                NaN  active
2162  2163       0           16   34.0  ...  NaN   NaN                NaN  active
2166  2167       0           18   37.0  ...  NaN   NaN                NaN  active
2167  2168       0           18  109.0  ...  NaN   NaN                NaN  active
...    ...     ...          ...    ...  ...  ...   ...                ...     ...
3870  3871       0           19   43.0  ...  NaN   NaN                NaN  active
3871  3872       0           19   43.0  ...  NaN   NaN                NaN  active
3872  3873       0           19   43.0  ...  NaN   NaN                NaN  active
3873  3874       0           19   43.0  ...  NaN   NaN                NaN  active
3874  3875       0           19   43.0  ...  NaN   NaN                NaN  active

[431 rows x 14 columns]

# Percent of active trucks. 

In [0]:
df2[df2['status']=='active']


id  pri_id  hs_customer  yard  ...  tag color licenseplatenumber  status
0        1     739            1   1.0  ...  NaN   NaN                NaN  active
1        2     740            1   1.0  ...  NaN   NaN                NaN  active
2        3     741            1   1.0  ...  NaN   NaN                NaN  active
3        4       0            2   3.0  ...  NaN   NaN                NaN  active
4        5       0            2   3.0  ...  NaN   NaN                NaN  active
...    ...     ...          ...   ...  ...  ...   ...                ...     ...
3870  3871       0           19  43.0  ...  NaN   NaN                NaN  active
3871  3872       0           19  43.0  ...  NaN   NaN                NaN  active
3872  3873       0           19  43.0  ...  NaN   NaN                NaN  active
3873  3874       0           19  43.0  ...  NaN   NaN                NaN  active
3874  3875       0           19  43.0  ...  NaN   NaN                NaN  active

[3241 rows x 14 columns]

In [0]:
df2[df2['status']=='inactive']

id  pri_id  hs_customer  ...  color  licenseplatenumber    status
19      20       0            5  ...  White             R130606  inactive
23      24       0            5  ...  White             R130613  inactive
24      25       0            5  ...  White             R130614  inactive
25      26       0            5  ...  White             R130615  inactive
26      27       0            5  ...  White             R130161  inactive
...    ...     ...          ...  ...    ...                 ...       ...
3309  3310       0           28  ...    NaN              706528  inactive
3346  3347       0           18  ...    NaN                 NaN  inactive
3372  3373       0           28  ...    NaN              2UJ789  inactive
3470  3471       0           28  ...    NaN                 NaN  inactive
3774  3775       0           18  ...    NaN                 NaN  inactive

[634 rows x 14 columns]

3241 active trucks. 634 inactive trucks. 19.56% of trucks are inactive

Customer that uses the most trucks.

In [0]:
count=df2['hs_customer'].value_counts()
print(count)
#cousomer 18 is the one using the most trucks.

18    904
36    693
25    586
19    215
16    137
45    121
20    121
5     118
32    116
17     77
21     76
34     62
12     54
41     51
43     50
35     47
4      47
6      46
10     37
11     36
15     35
8      35
23     33
13     26
28     25
39     23
9      18
1      17
40     14
37     13
38      9
3       9
2       9
27      5
33      5
22      3
14      1
44      1
Name: hs_customer, dtype: int64


In [0]:

df3 = df2[df2['status']=='active']


In [0]:
#customer 18 active trucks 734/904 |||| 170 trucks inactive, 18% of customers 18 trucks are inactive. 
df3[df3['hs_customer']==18]

id  pri_id  hs_customer   yard  ...  tag color licenseplatenumber  status
415    416       0           18   37.0  ...  NaN   NaN             701553  active
416    417       0           18   37.0  ...  NaN   NaN             700301  active
417    418       0           18   37.0  ...  NaN   NaN             700302  active
418    419       0           18   38.0  ...  NaN   NaN             700309  active
419    420       0           18   38.0  ...  NaN   NaN             702027  active
...    ...     ...          ...    ...  ...  ...   ...                ...     ...
3835  3836       0           18   46.0  ...  NaN   NaN                NaN  active
3836  3837       0           18   46.0  ...  NaN   NaN                NaN  active
3837  3838       0           18   46.0  ...  NaN   NaN                NaN  active
3863  3864       0           18   46.0  ...  NaN   NaN                NaN  active
3864  3865       0           18  134.0  ...  NaN   NaN                NaN  active

[734 rows x 14 columns]

how many unique truck id per customer 

In [9]:
df2[df2['hs_customer']==18]


id  pri_id  hs_customer   yard  ...  tag color licenseplatenumber  status
415    416       0           18   37.0  ...  NaN   NaN             701553  active
416    417       0           18   37.0  ...  NaN   NaN             700301  active
417    418       0           18   37.0  ...  NaN   NaN             700302  active
418    419       0           18   38.0  ...  NaN   NaN             700309  active
419    420       0           18   38.0  ...  NaN   NaN             702027  active
...    ...     ...          ...    ...  ...  ...   ...                ...     ...
3835  3836       0           18   46.0  ...  NaN   NaN                NaN  active
3836  3837       0           18   46.0  ...  NaN   NaN                NaN  active
3837  3838       0           18   46.0  ...  NaN   NaN                NaN  active
3863  3864       0           18   46.0  ...  NaN   NaN                NaN  active
3864  3865       0           18  134.0  ...  NaN   NaN                NaN  active

[904 rows x 14 columns]